src/libraries/System.Formats.Asn1/src/System/Formats/Asn1/WellKnownOids.cs

```diff
@@ -314,108 +115,211 @@ internal static class WellKnownOids
+                "1.2.840.113549.1.9.21" => data.Slice(361, 9), // LocalKeyId
+                "1.2.840.113549.1.9.22.1" => data.Slice(370, 10), // Pkcs12X509CertType
+                "1.2.840.113549.1.12.1.3" => data.Slice(380, 10), // Pkcs12TripleDes
+                "1.2.840.113549.1.12.1.5" => data.Slice(390, 10), // Pkcs12Rc2Cbc128
+                "1.2.840.113549.1.12.1.6" => data.Slice(400, 10), // Pkcs12Rc2Cbc40
+                "1.2.840.113549.1.12.10.1.1" => data.Slice(410, 11), // Pkcs12KeyBag
+                "1.2.840.113549.1.12.10.1.2" => data.Slice(421, 11), // Pkcs12ShroudedKeyBag
+                "1.2.840.113549.1.12.10.1.3" => data.Slice(432, 11), // Pkcs12CertBag
+                "1.2.840.113549.1.12.10.1.5" => data.Slice(443, 11), // Pkcs12SecretBag
+                "1.2.840.113549.1.12.10.1.6" => data.Slice(454, 11), // Pkcs12SafeContentsBag
+                "1.2.840.113549.2.5" => data.Slice(465, 8), // MD5
+                "1.2.840.113549.2.7" => data.Slice(473, 8), // HMACSHA1
+                "1.2.840.113549.2.9" => data.Slice(481, 8), // HMACSHA256
+                "1.2.840.113549.2.10" => data.Slice(489, 8), // HMACSHA384
+                "1.2.840.113549.2.11" => data.Slice(497, 8), // HMACSHA512
+                "1.2.840.113549.3.2" => data.Slice(505, 8), // RC2CBC
+                "1.2.840.113549.3.7" => data.Slice(513, 8), // TripleDESCBC
+                "1.3.6.1.4.1.311.17.1" => data.Slice(521, 9), // Pkcs12KeyProviderName
+                "1.3.6.1.4.1.311.17.3.20" => data.Slice(530, 10), // KeyIdentifier
+                "1.3.6.1.4.1.311.20.2.3" => data.Slice(540, 10), // UserPrincipalName
+                "1.3.6.1.4.1.311.88.2.1" => data.Slice(550, 10), // DocumentNameAttr
+                "1.3.6.1.4.1.311.88.2.2" => data.Slice(560, 10), // DocumentDescriptionAttr
+                "1.3.6.1.5.5.7.3.1" => data.Slice(570, 8), // KeyPurposeTlsServer
+                "1.3.6.1.5.5.7.3.2" => data.Slice(578, 8), // KeyPurposeTlsClient
+                "1.3.6.1.5.5.7.3.3" => data.Slice(586, 8), // KeyPurposeCodeSign
+                "1.3.6.1.5.5.7.3.4" => data.Slice(594, 8), // KeyPurposeEmailProtection
+                "1.3.6.1.5.5.7.3.8" => data.Slice(602, 8), // KeyPurposeTimestamping
+                "1.3.6.1.5.5.7.3.9" => data.Slice(610, 8), // KeyPurposeOcspSigner
+                "1.3.6.1.5.5.7.6.2" => data.Slice(618, 8), // Pkcs7NoSignature
+                "1.3.6.1.5.5.7.48.1" => data.Slice(626, 8), // OCSP
+                "1.3.6.1.5.5.7.48.1.2" => data.Slice(634, 9), // OcspNonce
+                "1.3.6.1.5.5.7.48.2" => data.Slice(643, 8), // CAIssuers
+                "1.3.14.3.2.26" => data.Slice(651, 5), // SHA1
+                "1.3.14.3.2.7" => data.Slice(656, 5), // DES
+                "1.3.132.0.34" => data.Slice(661, 5), // Secp384r1
+                "1.3.132.0.35" => data.Slice(666, 5), // Secp521r1
+                "2.5.4.3" => data.Slice(671, 3), // CommonName
```

Comment:
I'm a little bothered by how this will make future insertions hard to maintain in the same style.  Right now, everything is sorted by the dotted decimal string (or the binary encoding, since they sort the same way).  So, if in the future 2.5.4.2 is determined to be common enough to be worth encoding in the table, someone will have to decide between a) putting the new value at the end of the unified blob (being the first unsorted entry) or b) adding 3 to lines 300-321.

That said, adding things to this should be rare; so the savings might pay for the future frustration.